proto/storage/base_image_repository.proto

```diff
@@ -0,0 +1,71 @@
+syntax = "proto3";
+
+package storage;
+
+import "google/protobuf/timestamp.proto";
+
+option go_package = "./storage;storage";
+option java_package = "io.stackrox.proto.storage";
+
+// BaseImageRepository configures automatic tag discovery for a base image repository.
+// Defines which repository to watch, what tag patterns to match, and health tracking.
+// Next tag: 8
+message BaseImageRepository {
+  // Unique identifier for this repository pattern configuration.
+  string id = 1; // @gotags: sql:"pk,type(uuid)"
+
+  // Full repository path including registry (e.g., "registry.redhat.io/ubi8/ubi").
+  // Must be unique - one tag pattern per repository.
+  string repository_path = 2; // @gotags: sql:"unique"
+
+  // Glob pattern for tag filtering (e.g., "8.10-*").
+  string tag_pattern = 3;
+
+  // Timestamp of last successful poll.
+  google.protobuf.Timestamp last_poll_at = 4;
+
+  // Count of consecutive poll failures (for health tracking).
+  int32 failure_count = 5;
+
+  // Health status of this repository pattern.
+  enum HealthStatus {
+    HEALTHY = 0; // Pattern polling successfully.
+    UNHEALTHY = 1; // Pattern experiencing failures.
+    DISABLED = 2; // Pattern disabled by operator.
+  }
+  HealthStatus health_status = 6; // @gotags: sql:"index=btree"
+
+  // SHA256 hash of tag_pattern for cache invalidation detection
+  // (when pattern changes, hash differs, triggering cache flush).
+  string pattern_hash = 7;
+}
+
+// BaseImageTag stores cached base image tag metadata.
+// Persisting cache data across Central restarts avoids cold-start refetches during base
+// image repository scanning.
+// Next tag: 8
+message BaseImageTag {
+  // Unique identifier.
+  string id = 1; // @gotags: sql:"pk,type(uuid)"
+
+  // Foreign key to the repository pattern.
+  // Composite unique constraint with tag field.
+  string base_image_repository_id = 2; // @gotags: sql:"fk(BaseImageRepository:id),type(uuid),index=category:unique;name:base_image_tags_repo_tag"
```

Comment:
`base_image_tags_repo_tag`
Do you mean `base_image_tags_repo_id`?

Reply:
I am trying to make a composite unique field over the tuple (base_image_repository_id, tag), to ensure only one tag per repository ID exists in the database. Does that clarify my intention? Is my intention correctly expressed in the proto tags I used?